RC version 0.2.0

aburrell opened the pull request:
# Description

Release candidate fulfilling v0.2.0 milestones.

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality or documentation)
- This change requires a documentation update

# How Has This Been Tested?

Unit tests and examples

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant: RC branch of pysat

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on [readthedocs](https://pysatmadrigal--98.org.readthedocs.build/en/98/)
- [x] Upload the test distribution to the testPyPi server: https://test.pypi.org/project/pysatMadrigal/0.2.0rc1/
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the release text: https://github.com/pysat/pysatMadrigal/wiki/v0.2.0
- [x] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have linted the files updated in this pull request
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

aburrell committed:
BUG: fixed download stop time

Fixed the definition of the download stop time, not assuming it has a daily cadence.

aburrell committed:
ENH: added DMSP SSJ instrument

Added an instrument for the DMSP SSJ auroral boundary data.

aburrell committed:
DOC: updated the supported instruments

Added the DMSP SSJ instrument to the docs.

aburrell committed:
DOC: updated changelog

Added a summary of the changes to the changelog.

aburrell committed:
MAINT: removed temporary code

Removed code needed to handle failure of pysat to allow local definitions of start/stop times.

aburrell committed:
DOC: updated CHANGELOG

Updated changelog with a description of this pull.

aburrell committed:
MAINT: updated pysat version limit

Updated the pysat version limit to the next release that will include all the necessary changes for pysatMadrigal to work as expected.

aburrell committed:
Merge pull request #85 from pysat/rc_v0.1.0

Rc v0.1.0

jklenzing committed:
TST: add rc install workflow

jklenzing committed:
MAINT: update pip syntax

jklenzing committed:
DOC: update changelog

jklenzing committed:
MAINT: set to push

jklenzing committed:
TST: set to workflow_dispatch

jklenzing committed:
Update pip_rc_install.yml

jklenzing committed:
Update test_requirements.txt

jklenzing committed:
ENH: add version to package

jklenzing committed:
DOC: add docstring

aburrell committed:
MAINT: remove ssj from general pandas

Because DMSP SSJ requires annual loading, it cannot be handled correctly by the general pandas instrument.

jklenzing committed:
Merge pull request #86 from pysat/tst/ga_updates

MAINT: GA updates

aburrell committed:
BUG: add sphinx version limit

Fix CI docs tests with a sphinx version limit.

aburrell committed:
STY: made flake8 updates

Updated the docstring and import style.

aburrell committed:
STY: update flake8

Attempt to update flake8 so PEP8 tests pass.

aburrell committed:
STY: removed extra blank line

Removed an extra blank line in a docstring.

aburrell committed:
BUG: fixed variable case

Fixed variable names to lowercase, to comply with pysat standards.

aburrell committed:
Merge pull request #81 from pysat/dmsp_ssj4

ENH: Added a DMSP SSJ instrument

aburrell committed:
Merge pull request #87 from pysat/dmsp_ssj4

Recent changes update

aburrell committed:
Merge pull request #82 from pysat/update_pysat_remote_file_list

Update pysat remote file list

aburrell committed:
TST: added windows to CI

Added Windows to CI, may or may not work.

aburrell committed:
TST: added close statement

Added a close statement to help Windows OS.

aburrell committed:
BUG: removed the AE data

Removed the AE data from the general madrigal instrument, as they have a very large file and by default have duplicated times.

aburrell committed:
BUG: allow any kwarg input to general download

Allow any kwarg input to the general download function, improving ease of use with functools.

aburrell committed:
ENH: added the NGDC AE instrument

Added an instrument for the NGDC AE data.  This should fix the windows memory issue, as it only loads the desired portion of the data (which is all contained in a single file).

aburrell committed:
DOC: updated changelog

Added a summary of the changes to the changelog.

aburrell committed:
DOC: updated supported instruments

Added the new NGDC AE instrument to the documentation.

aburrell committed:
STY: added missing period

Added a missing period the the module docstring.

aburrell committed:
BUG: clearer logic statement

Make the logic statement clearer.

aburrell committed:
STY: fixed line length

Fixed the line length for a GitHub suggestion.

aburrell committed:
BUG: fixed test bug

Fixed the test bug introduced by trying to be clever.

jklenzing committed:
Update pip_rc_install.yml

aburrell committed:
BUG: fixed string evalutation

Fixed a string evaluation that should have included an equality.  Also removed a logger warning for duplicate times, as this should be caught by pysat.

aburrell committed:
Merge pull request #89 from pysat/211_inst

NGDC AE (211) Instrument

aburrell committed:
BUG: test the correct variable names

Changed from testing 'min' to 'hm' and 'sec' to 'hmi', as these are the time variables needed to create a uniquely defined timestamp for this file.

aburrell committed:
Merge pull request #92 from pysat/211_inst

211 inst

aburrell committed:
BUG: added try/except loop

Added a pass-catch for the Windows PermissionError.

aburrell committed:
BUG: bad keyboard

Removed extra letters caused by my laptop's broken keyboard.

aburrell committed:
BUG: changed try/except location

Moved the try/except loop.

aburrell committed:
BUG: file error popping up again

Windows just hates closing files.  Added more close and try/except statements.

aburrell committed:
STY: fixed TODO comment

Fixed the TODO formatting.

aburrell committed:
BUG: allow many types of errors

Windows will throw many things at you to avoid removing data.

aburrell committed:
TST: remove unused close call

Files do not have the `close` method, remove useless call.

aburrell committed:
STY: changed wording

Changed wording without any change in meaning or clarity to satisfy @jklenzing.

Co-authored-by: Jeff Klenzing <19592220+jklenzing@users.noreply.github.com>

aburrell committed:
Merge pull request #88 from pysat/windows_ci

TST: added windows to CI

aburrell committed:
MAINT: added Windows to configuration

Added the fact that we now support Windows to the setup.cfg file.

aburrell committed:
TST: added `_clean_warn` attribute

Added the `_clean_warn` attribute to all existing Instruments.

aburrell committed:
DOC: update Changelog

Updated the changelog with a description of the PR.

aburrell committed:
Merge pull request #93 from pysat/clean_warn_update

Clean warn update

aburrell committed:
ENH: removed IMF data from general inst

Removed the IMF data from the general pandas instrument, since it is all contained in one file and requires special handling.

aburrell committed:
DOC: updated supported instruments page

Added the OMNI IMF instrument to the supported instruments page.

aburrell committed:
DOC: updated changelog

Added a description of the current changes to the log.

aburrell committed:
DOC: update AE docstring

Fixed the AE module docstring so that the example doesn't wrongly reference DMSP.

aburrell committed:
ENH: added index instruments

Added Instruments for the Dst, IMF, and geomagnetic indices provided by Madrigal.

aburrell committed:
MAINT: removed support for single-file data sets

Removed general support for the index data sets, which require special processing because all data is contained in a single file.

aburrell committed:
BUG: fixed file times

Fixed the file time range determination in the load function.  Also updated the docstrings to be more correct.

aburrell committed:
DOC: added new instruments to docs

Added the new index instruments to the docs.

aburrell committed:
DOC: updated changelog

Added the other instruments to the changelog.

aburrell committed:
STY: fixed import order

Fixed the import order to be alphabetical.

aburrell committed:
MAINT: updated JRO link

Updates the link to the Jicamarca Radio Observatory.

aburrell committed:
Merge pull request #94 from pysat/remove_gen_pandas_special

Move Madrigal index Instruments to their own sub-modules

aburrell committed:
DOC: update issue templates

Fixed typos in the issue templates.

aburrell committed:
DOC: updated PR template

Added new item for the PR checklist.

aburrell committed:
TST: updated test yamls

Updated the test yamls to use pyproject.toml and cycle the version support.

aburrell committed:
ENH: added readthedocs yaml

Added a readthedocs yaml.

aburrell committed:
DOC: updated code of conduct

Added missing section to the code of conduct.

aburrell committed:
DOC: updated contributing guidelines

Fixed format and updated style requirements in the contributing guidelines.

aburrell committed:
MAINT: added pyproject.toml

Replaced the setup.py with pyproject.toml.

aburrell committed:
DOC: updated README

Fixed broken links and updated dependencies in the README.

aburrell committed:
MAINT: updated version support

Updated the version support to use pyproject.toml.

aburrell committed:
DOC: updated changelog

Added a summary of the changes to the changelog.

aburrell committed:
BUG: fixed if statement in main.yml

Fixed matrix variables so that new if statement is triggered.

aburrell committed:
DOC: updated conf.py

Updated version access in conf.py.

aburrell committed:
DOC: removed duplicate links

Removed duplicate explicit links.

aburrell committed:
DOC: updated example index

Included a missing example in the example index.

aburrell committed:
BUG: removed comma

Removed comma from authors.

aburrell committed:
DOC: updated installation guide

Update the installation guide to reflect the change to pyproject.toml.

aburrell committed:
DOC: updated Manifest

Removed the old version.txt file from the manifest.

aburrell committed:
Merge pull request #95 from pysat/update_templates

MAINT: Update templates

aburrell committed:
STY: refactored general load function

Extracted useful elements of the `load` function to allow their use elsewhere without code duplication.

aburrell committed:
ENH: added gnss load functions

Added type-specific GNSS load functions and exploratory file functions for the line-of-sight files.

aburrell committed:
ENH: added 'los' tag to gnss_tec

Added support for the line-of-sight (los) TEC data.

aburrell committed:
DOC: updated GNSS method description

Improved the GNSS instrument method description to encompass the new scope.

aburrell committed:
DOC: updated changelog

Added a summary of the changes to the changelog.

aburrell committed:
STY: fixed PEP8 violations

Removed excess lines, fixed typos, added missing imports.

aburrell committed:
DOC: updated docstring

Added missing return value to the docstring.

aburrell committed:
BUG: added load options for los data

Added two load options for the GNSS TEC LoS tests.

aburrell committed:
BUG: allow empty data

Allow creation of an empty Dataset from an empty DataFrame.

aburrell committed:
BUG: allow empty file list

Allow an empty filelist to be provided and create an empty data object.

aburrell committed:
BUG: allow empty gnss load

Allow the GNSS TEC load to work with an empty file list.

aburrell committed:
TST: added load tests

Added load tests for the LoS logger warning and ValueError.

aburrell committed:
STY: fixed PEP8 errors

Removed unused variable and fixed indentation.

aburrell committed:
TST: skip los CI download tests

Corrected the download test flag.

Co-authored-by: Jeff Klenzing <19592220+jklenzing@users.noreply.github.com>

aburrell committed:
BUG: fixed clean method

Fixed the clean method to have an informative message for the LoS data.  Also removed inaccessable line in the load method.

aburrell committed:
TST: added gnss_tec load error test

Added a test for the gnss_tec load method ValueError raised if a bad `los_value` is supplied.

aburrell committed:
TST: added general unit tests

Added unit tests for uncovered lines in the general instruments methods sub-module.  Also improved style for Error tests.

aburrell committed:
TST: added empty load test

Added a test for loading no data in the `load_los` function, additionally testing the time selection set-up.

aburrell committed:
TST: added coveralls line

Added a coveralls service job environment variable.

aburrell committed:
BUG: accessed correct instrument kwargs

Needed to get the right dict from the setup.

aburrell committed:
BUG: add fake files

Add a fake file to test the load failures.

aburrell committed:
STY: fixed import order

Fixed import order to be alphabetical.

aburrell committed:
TST: updated coveralls

Updated coveralls to run in parallel.

aburrell committed:
TST: disabled local LoS download tests

Disabled local LoS download tests, also added a warning for users attempting downloads.

aburrell committed:
TST: removed old test line

Removed old test line.

aburrell committed:
MAINT: updated version caps

Updated the version caps for coveralls, sphinx, and sphinx_rtd_theme.

aburrell committed:
Revert "TST: removed old test line"

This reverts commit 9b8919b5311d837f384b86a1a1c9c2b702a7b29e.

aburrell committed:
BUG: removed bad test variable

Removed a test variable that was being used for an incorrect purpose.

aburrell committed:
BUG: fixed workflow

Removed extra 'run' from workflow.

aburrell committed:
MAINT: updated coveralls

Updated the coveralls line in the RC CI test.

aburrell committed:
BUG: updated DMSP url

Added the 'url' kwarg to the DMSP IVM download method, since UTD downloads were failing at the central site.

aburrell committed:
BUG: removed url kwarg

Removed the URL kwarg, local tests didn't need the url change.

aburrell committed:
BUG: added missing index

Added a missing index for the test list.

aburrell committed:
Merge pull request #96 from pysat/los_tec

Added LoS TEC tag to the GNSS TEC Instrument

aburrell committed:
REL: update version number

Update version number in preparation for release.

aburrell committed:
REL: update changelog

Update the changelog release version number and target date.

aburrell committed:
DOC: add acknowledgements

Add acknowledgements to the documentation and the main directory.

aburrell committed:
DOC: update file headers

Update the code file headers to include the NRL pub release.

aburrell committed:
MAINT: updated readme

Updated the readme install instructions and fixed a typo in the example.

aburrell committed:
DOC: Update CHANGELOG.md

Update the targeted release date to unspecified in preparation for a separate release candidate branch.

aburrell committed:
MAINT: update acknowledgements

Update the acknowledgements.

Co-authored-by: Jeff Klenzing <19592220+jklenzing@users.noreply.github.com>

aburrell committed:
Merge pull request #97 from pysat/nrl_pub_release

NRL pub release

aburrell committed:
TST: updated CI yamls

Updated the CI yamls by:
- upgrading the action versions,
- adding support for Python 3.12, and
- cycling versions performing doc/pypi tests.

aburrell committed:
MAINT: updated the python version

Updated the supported python versions.

aburrell committed:
DOC: set target release date

Set a target release date in the changelog.  Also added maintenance details.

aburrell committed:
DOC: change index depth

Change the index depth to have a maximum of two.

jklenzing commented on pysatMadrigal/instruments/dmsp_ivm.py:
```diff
@@ -104,6 +105,13 @@
     'f17': {tag: dt.datetime(2009, 1, 1) for tag in inst_ids['f17']},
     'f18': {tag: dt.datetime(2017, 12, 30) for tag in inst_ids['f18']}}
 
+_clean_warn = {inst_id: {tag: {clean_level: [('logger', 'WARN',
```
```suggestion
# TODO(#99): Remove when compliant with multi-day load tests
_new_tests = {'f18': {'': False}}
_clean_warn = {inst_id: {tag: {clean_level: [('logger', 'WARN',
```

jklenzing commented on pyproject.toml:
```diff
@@ -0,0 +1,90 @@
+  "all_inst",
+  "download",
+  "no_download",
+  "load_options",
```
```suggestion
  "load_options",
  "new_tests",
```

aburrell committed:
BUG: bug fixes for pysat RC

Fixes tests that are failing in the pysat RC.

Co-authored-by: Jeff Klenzing <19592220+jklenzing@users.noreply.github.com>

aburrell committed:
DOC: added comment

Added a comment to clarify which part is involved in the TODO statement.

aburrell committed:
TST: added skip for 'drifts' and 'drifts_ave'

The multi-day load tests are failing for the drift data, this may be a test day issue.

aburrell committed:
TST: extend JRO new test skipping

Extending the JRO ISR tags that skip the new tests.

aburrell committed:
BUG: added flag for kindat 7800

Added a flag to not perform the new tests with the general pandas Madrigal instrument for the 7800 dataset.

aburrell committed:
BUG: fixed bug in warn dict

Fixed a badly named variable in the clean warning definition dict.

aburrell committed:
BUG: second fix for clean warnings

Test failing due to missing key, try empty list.

aburrell committed:
BUG: fixed AE clean warning

Fixed the expected AE clean warning.

aburrell committed:
STY: removed trailing whitespace

Removed trailing whitespace from a line.

aburrell committed:
BUG: for debugging, removing test flag

Temporarily remove test flag for RC tests.

aburrell committed:
TST: update test flags

Add another inst_id/tag combo to the skips.

aburrell committed:
TST: update JRO methods tests

Update the test Instrument used in the JRO methods test.

aburrell committed:
TST: re-added quick fail

Added the quick fail back into the tests.

rstoneback commented on pysatMadrigal/instruments/madrigal_pandas.py:
```diff
@@ -75,7 +75,9 @@
-excluded_tags = ['8105']  # Pandas-style data that requires special support
+
+# Pandas-style data that requires special support
+excluded_tags = ['120', '180', '210', '211', '212', '8105']
```
May want to consider some form of message to the user if they try to load an excluded tag. In the code changes highlighting I don't see where this excluded_tag gets used.

aburrell replied:
Look a few lines down (87).  They get the standard python "unknown tag" warning, like any other instrument.

rstoneback commented on pysatMadrigal/instruments/madrigal_pandas.py:
```diff
@@ -235,36 +241,39 @@ def list_files(tag, inst_id, data_path, kindat='', format_str=None,
+        If user or password are not supplied
+
+    Note
+    ----
```
Checking in since I believe that user names previously needed the '+' in between.

aburrell replied:
That was an old convention, which isn't necessary.

rstoneback commented on pysatMadrigal/instruments/methods/general.py:
```diff
@@ -481,13 +483,239 @@ def madrigal_file_format_str(inst_code, strict=False, verbose=True):
+        if fname not in load_file_types[ftype]:
+            # Raise a logger warning if a file with an unknown extension
+            # is encountered
+            pysat.logger.warning(
```
I wonder if this should be a warnings.warning instead. For your consideration since users can do something about this situation.

aburrell replied:
I'd be fine with that on a future version, since we'd need to add a test.  Maybe make an issue?

rstoneback commented on pysatMadrigal/instruments/methods/general.py:
```diff
@@ -481,13 +483,239 @@ def madrigal_file_format_str(inst_code, strict=False, verbose=True):
+
+
+def convert_pandas_to_xarray(xarray_coords, data, time_ind):
+    """Convert Madrigal HDF5/simple data from pandas to xarray.
```
It says simple but the code implies otherwise :)

rstoneback commented on pysatMadrigal/instruments/methods/general.py:
```diff
@@ -671,120 +874,22 @@ def load(fnames, tag='', inst_id='', xarray_coords=None):
-                                            "consider specifing additional",
-                                            " coordinates and storing the ",
-                                            "data as an xarray Dataset"]))
+                    pysat.logger.warning(''.join([
```
This may be a warnings.warning situation as well.

rstoneback commented on pysatMadrigal/instruments/ngdc_ae.py:
```diff
@@ -0,0 +1,303 @@
+def init(self):
+    """Initialize the Instrument object in support of Madrigal access.
+
+    Parameters
```
noting 'kindat' again in init in case there is a change.

rstoneback commented on pysatMadrigal/instruments/ngdc_ae.py:
```diff
@@ -0,0 +1,303 @@
+
+    Note
+    ----
+    Supports 'clean', 'dusty', 'dirty' in the sense that it prints
+    a message noting there is no cleaning.
+    'None' is also supported as it signifies no cleaning.
```
There is cleaning in this method.

rstoneback commented on pysatMadrigal/instruments/omni2_imf.py:
```diff
@@ -0,0 +1,235 @@
+# DISTRIBUTION STATEMENT A: Approved for public release. Distribution is
+# unlimited.
+# ----------------------------------------------------------------------------
+"""Supports access to OMNI 2 IMF data archieved at Madrigal.
```
```suggestion
"""Supports access to OMNI 2 IMF data archived at Madrigal.
```

rstoneback commented on pysatMadrigal/instruments/omni2_imf.py:
```diff
@@ -0,0 +1,235 @@
+
+    Parameters
+    ----------
+    kindat : str
```
kindat note.

rstoneback commented on pysatMadrigal/instruments/omni2_imf.py:
```diff
@@ -0,0 +1,235 @@
+    if data.index[-1] > madrigal_end:
+        pysat.logger.critical(''.join(['There is data beyond ',
+                                       '{:}'.format(madrigal_end), ' in the ',
+                                       'Omni2 IMF file, please notify the ',
```
```suggestion
                                       'OMNI2 IMF file, please notify the ',
```

rstoneback commented on pysatMadrigal/instruments/omni2_imf.py:
```diff
@@ -0,0 +1,235 @@
+                                       '{:}'.format(madrigal_end), ' in the ',
+                                       'Omni2 IMF file, please notify the ',
+                                       'pysatMadrigal developers so that they ',
+                                       'can update this Instrument']))
```
```suggestion
                                       'can update this Instrument.']))
```

rstoneback commented on pysatMadrigal/tests/test_instruments.py:
```diff
@@ -1,12 +1,22 @@
 #!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
First pass on review, made it to here.

rstoneback reviewed (commented):
First pass through, review not quite complete. Release is looking good!

rstoneback commented on pysatMadrigal/tests/test_methods_general.py:
```diff
@@ -323,21 +342,29 @@ def test_download_valueerror(self, in_key, in_val, test_verr):
+        return
 
-        assert str(verr).find("unknown date_array supplied") >= 0
+    def test_convert_pandas_to_xarray_bad_data_vars(self):
```
May want to consider additional testing for the underlying method. Seems like pandas to xarray could have a variety of challenges.

aburrell replied:
Please make this an issue.

rstoneback commented on pysatMadrigal/tests/test_methods_gnss.py:
```diff
@@ -31,3 +39,49 @@ def test_ref_output(self, func, comp_str, in_args):
+        with caplog.at_level(logging.WARN, logger='pysat'):
+            gnss.load_los(self.bad_fnames, "site", "zzon")
+
+        # Test the logger warning
```
May want to use the pysat.testing warnings methods.

aburrell replied:
Since you've identified places, recommend creating an issue and adding TODO statements at the locations you found.

rstoneback reviewed (commented):
Second pass complete. Turns out I was close to the end first time round.

aburrell commented on pysatMadrigal/instruments/dmsp_ivm.py:
```diff
@@ -104,6 +105,18 @@
+_clean_warn = {inst_id: {tag: {clean_level: [('logger', 'WARN',
+                                              'this level 1 data has no qual',
+                                              clean_level)]
+                               if tag != 'utd' else []
+                               for clean_level in ['clean', 'dusty', 'dirty']}
+                         for tag in inst_ids[inst_id]}
+               for inst_id in inst_ids.keys()}
```
It's a nest of dicts.

aburrell commented on pysatMadrigal/instruments/ngdc_ae.py:
```diff
@@ -0,0 +1,303 @@
+
+
+def init(self):
+    """Initialize the Instrument object in support of Madrigal access.
+
+    Parameters
+    ----------
+    kindat : str
+        Madrigal instrument experiment code(s). (default='')
+
+    """
```
```suggestion
    """Initialize the Instrument object in support of Madrigal access."""
```

aburrell commented on pysatMadrigal/instruments/ngdc_ae.py:
```diff
@@ -0,0 +1,303 @@
+
+    Note
+    ----
+    Supports 'clean', 'dusty', 'dirty' in the sense that it prints
+    a message noting there is no cleaning.
```
```suggestion
    Supports 'clean', 'dusty', 'dirty' in the sense that all
    levels use the flag to clean data the same way.
```

aburrell commented on pysatMadrigal/instruments/omni2_imf.py:
```diff
@@ -0,0 +1,235 @@
+
+
+def init(self):
+    """Initialize the Instrument object in support of Madrigal access.
+
+    Parameters
+    ----------
+    kindat : str
+        Madrigal instrument experiment code(s). (default='')
+
+    """
```
```suggestion
    """Initialize the Instrument object in support of Madrigal access."""
```

aburrell commented on pysatMadrigal/tests/test_methods_general.py:
```diff
@@ -42,6 +47,25 @@ def test_acknowledgements(self):
+        pysat.utils.testing.assert_lists_equal(list(self.out.keys()),
+                                               ['hdf5', 'netCDF4', 'simple'])
+
+        # Evaluate the logger warning
```
Can you make this an issue?

aburrell committed:
MAINT: applied suggestions

Applied suggestions from code review.

Co-authored-by: Russell Stoneback <github@stoneris.com>

aburrell commented on pysatMadrigal/instruments/dmsp_ssj.py:
```diff
@@ -0,0 +1,275 @@
+    self.acknowledgements = ''.join([
+        general.cedar_rules(), '\nThe Air Force Research Laboratory Auroral ',
+        'Boundary Index (ABI) was provided by the United States Air Force ',
+        'Research Laboratory, Kirtland Air Force Base, New Mexico.'])
```
```suggestion
        'Research Laboratory, Kirtland Air Force Base, New Mexico.\n',
        'Please send a copy of all publications that use the ABI to Dr. ',
        'Gordon Wilson at: gordon.wilson@kirtland.af.mil\n',
        'Dr. Wilson considers this data set deprecated.'])
```

aburrell commented on pysatMadrigal/instruments/dmsp_ssj.py:
```diff
@@ -0,0 +1,275 @@
+
+    Note
+    ----
+    Currently HDF5 reading breaks if a different file type was used previously
```
I read it as only try to download one version of this file.

aburrell commented on pysatMadrigal/instruments/dmsp_ssj.py:
```diff
@@ -0,0 +1,275 @@
+
+    Note
+    ----
+    The user's names should be provided in field user. Ritu Karidhal should
```
I looked up different space physicists to use.

aburrell commented on pysatMadrigal/instruments/dmsp_ssj.py:
```diff
@@ -0,0 +1,275 @@
+
+    """
+    # Ensure the date range is correct
+    if date_array.freq not in ['AS-JAN', 'YS', 'AS']:
```
We don't do that in most places.  If we do, it will need to be a pysat-wide thing.

aburrell committed:
DOC: improve DMSP SSJ init

Improve the user output for the DMSP SSJ instrument.

aburrell commented on pysatMadrigal/instruments/gnss_tec.py:
```diff
@@ -239,47 +293,53 @@ def load(fnames, tag='', inst_id=''):
+    elif tag == 'los':
+        if los_value is None:
+            raise ValueError('must specify a valid {:}'.format(los_method))
+
```
I don't understand what you're asking for.  Can you make it a suggestion, so I know exactly what you want?

aburrell commented on pysatMadrigal/instruments/madrigal_dst.py:
```diff
@@ -0,0 +1,212 @@
+
+# Local attributes
+#
+# Need a way to get the filename strings for a particular instrument unless
```
They only work in specific places, as noted in the pysat code.

aburrell commented on pysatMadrigal/instruments/madrigal_dst.py:
```diff
@@ -0,0 +1,212 @@
+    Raises
+    ------
+    ValueError
+        Unexpected time variable names
```
It's helpful if the files change remotely, so the user knows where it went wrong and can give a useful issue/bug report.

aburrell commented on pysatMadrigal/instruments/madrigal_geoind.py:
```diff
@@ -0,0 +1,228 @@
+
+
+def init(self):
+    """Initialize the Instrument object in support of Madrigal access.
+
+    Parameters
+    ----------
+    kindat : str
+        Madrigal instrument experiment code(s). (default='')
+
+    """
```
```suggestion
    """Initialize the Instrument object in support of Madrigal access."""
```

aburrell commented on pysatMadrigal/instruments/madrigal_geoind.py:
```diff
@@ -0,0 +1,228 @@
+
+
+def load(fnames, tag='', inst_id=''):
+    """Load the Madrigal Dst data.
```
```suggestion
    """Load the Madrigal geoindex data.
```

aburrell commented on pysatMadrigal/instruments/madrigal_geoind.py:
```diff
@@ -0,0 +1,228 @@
+        Unexpected time variable names
+
+    """
+    # Cycle through all the filenames, getting the desired start and stop times
```
This would make a good issue.

aburrell commented on pysatMadrigal/instruments/madrigal_geoind.py:
```diff
@@ -0,0 +1,228 @@
+    Returns
+    --------
+    data : pds.DataFrame
+        Object containing IMF data
```
```suggestion
        Object containing geoindex data
```

aburrell committed:
MAINT: applied code review

Found more code review comments and made changes.

Co-authored-by: Russell Stoneback <github@stoneris.com>

rstoneback commented on pysatMadrigal/tests/test_methods_gnss.py:
```diff
@@ -31,3 +39,49 @@ def test_ref_output(self, func, comp_str, in_args):
+        with caplog.at_level(logging.WARN, logger='pysat'):
+            gnss.load_los(self.bad_fnames, "site", "zzon")
+
+        # Test the logger warning
```
```suggestion
        # Test the logger warning
        # TODO(#101) Use pysat eval warnings
```

rstoneback commented on pysatMadrigal/tests/test_methods_general.py:
```diff
@@ -42,6 +47,25 @@ def test_acknowledgements(self):
+        pysat.utils.testing.assert_lists_equal(list(self.out.keys()),
+                                               ['hdf5', 'netCDF4', 'simple'])
+
+        # Evaluate the logger warning
```
```suggestion
        # Evaluate the logger warning
        # TODO(#101) Use pysat warnings test
```

rstoneback commented on pysatMadrigal/tests/test_methods_general.py:
```diff
@@ -323,21 +342,29 @@ def test_download_valueerror(self, in_key, in_val, test_verr):
+        return
 
-        assert str(verr).find("unknown date_array supplied") >= 0
+    def test_convert_pandas_to_xarray_bad_data_vars(self):
```
```suggestion
    # (TODO #102) Expand testing of pandas to xarray method below
    def test_convert_pandas_to_xarray_bad_data_vars(self):
```

aburrell committed:
DOC: add TODO statements

Added TODO statements for new issues.

Co-authored-by: Russell Stoneback <github@stoneris.com>

aburrell commented on pysatMadrigal/tests/test_methods_general.py:
```diff
@@ -323,21 +343,30 @@ def test_download_valueerror(self, in_key, in_val, test_verr):
+        return
 
-        assert str(verr).find("unknown date_array supplied") >= 0
+    # (TODO #102) Expand testing of pandas to xarray method below
```
```suggestion
    # TODO(#102) Expand testing of pandas to xarray method below
```

aburrell committed:
BUG: fix TODO formatting

Use the correct TODO formatting.

aburrell commented on .github/workflows/main.yml:
```diff
@@ -12,35 +12,47 @@ jobs:
-        python-version: ["3.9", "3.10"]
-        numpy_ver: ["latest"]
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.10", "3.11", "3.12"]
```
```suggestion
        python-version: ["3.10", "3.11"]
```

aburrell commented on pyproject.toml:
```diff
@@ -0,0 +1,91 @@
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
```
```suggestion
```

aburrell committed:
REV: python 3.12 failing on mac

Remove support for 3.12 from this release.

aburrell commented:
Poking @rstoneback and @JonathonMSmith for a review!

rstoneback commented:
I've been trying to sort out where the reduction in testing coverage came from but coveralls has been giving me gateway errors for a couple days. Still is. Don't want to keep holding things up.

aburrell commented:
@rstoneback it's because we can't download the LoS data (it's too big to handle through GH Actions).  Coveralls has been stretched lately 😿

aburrell commented on CHANGELOG.md:
```diff
@@ -3,6 +3,25 @@ Change Log
+* Maintenance
+  * Add manual GitHub Actions tests for pysatMadrigal RC
+  * Update GitHub Actions workflows for newer versions of pip, updated actions
+  * Added support for Python 3.12
```
```suggestion
```

aburrell reviewed (commented):
Found a line in the changelog that needs to be fixed.

aburrell committed:
MAINT: Update CHANGELOG.md

Support for python 3.12 was removed until a documented issue can be solved.
